File: Conexion base de datos/GO/MySql.go
package main

import (
	"database/sql"
	"fmt"
	"log"

	_ "github.com/go-sql-driver/mysql" // Driver MySQL
)

func main() {
	// Configuración de conexión (similar a tu ejemplo Python)
	dsn := "root:123456@tcp(localhost:3306)/" // Formato: usuario:contraseña@protocolo(host:puerto)/basedatos

	// Abrir conexión
	db, err := sql.Open("mysql", dsn)
	if err != nil {
		log.Fatal("Error al abrir conexión:", err)
	}
	defer db.Close() // Importante: cerrar conexión al final

	// Verificar conexión
	err = db.Ping()
	if err != nil {
		log.Fatal("Error al conectar:", err)
	}

	fmt.Println("✅ Conexión exitosa a MySQL")
}

File: Conexion base de datos/JAVASCRIPT/MySql.js

const mysql = require('mysql2/promise');  

async function connectMySQL() {
    let connection;
    
    try {
        
        connection = await mysql.createConnection({
            host: 'localhost',
            user: 'root',
            password: '123456'
        });

        console.log('✅ Conexión exitosa a MySQL');

        const [rows, fields] = await connection.execute('SHOW DATABASES');
        console.log('Bases de datos disponibles:', rows.map(row => row.Database));

    } catch (err) {
        console.error('❌ Error de conexión:', err.message);
    } finally {
        
        if (connection) await connection.end(); 
    }
}

connectMySQL();